src/components/Table/Table.tsx

```diff
@@ -1,27 +1,85 @@
-import { Table as MantineTable, TextInput } from "@mantine/core";
-import { type ChangeEvent, type ReactNode, useMemo, useState } from "react";
-import { IconSortAscending, IconSortDescending } from "@tabler/icons-react";
+import {
+  Table as MantineTable,
+  TextInput,
+  Pagination as MantinePagination,
+  createStyles,
+  Box,
+} from "@mantine/core";
+import {
+  type ChangeEvent,
+  type ReactNode,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import {
+  IconArrowBarToLeft,
+  IconArrowBarToRight,
+  IconArrowLeft,
+  IconArrowRight,
+  IconGripHorizontal,
+  IconSortAscending,
+  IconSortDescending,
+} from "@tabler/icons-react";
 import { omit } from "lodash/fp";
-import { filterData, getNextSortValue, sortData } from "../../utils/utils.ts";
+import {
+  filterData,
+  getMaximumPages,
+  getNextSortValue,
+  sortData,
+} from "../../utils/utils.ts";
+import { useElementSize } from "@mantine/hooks";
 
 export interface Column<TData> {
   columnName: string;
   exactMatch: boolean;
   header: string;
   cellRenderer?: (row: TData) => ReactNode;
+  isFilterable: boolean;
 }
 
+interface Pagination {
+  pageSize: number;
+}
 interface TableProps<TData> {
   data: TData[];
   columns: Array<Column<TData>>;
+  pagination: Pagination;
 }
 
 export type Sort = Record<string, "asc" | "desc" | null>;
 export type Filter = Record<string, string>;
 
 export function Table<T extends { id: number }>(props: TableProps<T>) {
+  const { classes } = useStyles();
+
+  const { ref: rowRef, height: rowHeight } = useElementSize();
+
+  const [windowHeight, setWindowDimensions] = useState(window.innerHeight);
+  const [numberOfRows, setNumberOfRows] = useState<number>(
+    props.pagination.pageSize
+  );
+
   const [filtersState, setFilters] = useState<Filter>({});
   const [sortState, setSorting] = useState<Sort>({});
+  const [paginationResults, setPaginationResults] = useState<number>(1);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowDimensions(window.innerHeight);
+      // -5 rows to account for the header, pagination, padding and filter row
```

Comment:
Always try making exact calculations, and not "~5 rows ~= header + pagination + padding`

1. Move the padding size to a const -> `const PADDING = x;`
2. Set a fixed height for pagination & headers elements, move to a const -> `const HEADER_HEIGHT = x; const PAGINATION_HEIGHT = x;`
3. Use all of the above for calculation of the height: `windowHeight - PADDING - HEADER_HEIGHT - PAGINATION_HEIGHT`

4. All of this will be redundant when you base your logic on the Table's height, instead of the window